# Assignment 1

In [ ]:
#Author-> Sanath Shivaswamy
#ID -> W1166731

In [2]:
import pandas as pd

In [3]:
df = pd.read_csv("data/eu_revolving_loans.csv", 
                 header=[1,2,3], index_col=0, skiprows=1, na_values=['-'])

In [4]:
df.columns = df.columns.droplevel(1)
df.columns = df.columns.droplevel(1)
df.head()

Austria  Belgium  Cyprus  Germany  Estonia  Spain  Finland  France  \
2016Feb     4.20     8.09    6.13     8.85    16.38  11.03     6.41    6.00   
2016Jan     4.17     8.05    6.13     8.83    16.41  10.85     6.44    6.10   
2015Dec     3.99     8.21    6.16     8.69    16.32  10.49     6.40    5.86   
2015Nov     4.09     8.20    6.15     8.82    16.37  10.73     6.49    6.18   
2015Oct     4.10     8.40    6.16     8.89    16.42  10.86     6.52    6.15   

         Greece (GR)  Ireland  Italy  Lithuania  Luxembourg  Latvia  Malta  \
2016Feb         8.18    10.40   6.07       9.60        7.24   21.58   6.79   
2016Jan         8.19     9.91   6.07       9.58        7.10   21.51   6.82   
2015Dec         8.22    10.37   5.91       9.45        6.96   21.59   6.84   
2015Nov         8.30     9.11   6.12       9.53        6.88   21.67   6.78   
2015Oct         8.34     9.29   6.11       9.40        7.08   21.74   6.78   

         Netherlands  Portugal  Slovenia  Slovakia  \
2016Feb         5.83     15.28      6.80     13.86   
2016Jan         5.81     15.41      6.79     13.97   
2015Dec         5.88     15.16      6.65     13.72   
2015Nov         5.97     15.32      6.79     14.10   
2015Oct         5.93     15.55      6.74     14.12   

         Euro area (changing composition)  
2016Feb                              7.36  
2016Jan                              7.35  
2015Dec                              7.23  
2015Nov                              7.37  
2015Oct                              7.41

### 1) Which EU Country has the largest revolving loan % increase from 2000 – 2016?

In [6]:
### Largest Percent Increase -> Latvia
for country in df: #Looping through each country
    df_temp = df[country].dropna() #drop NaN values for that country
    percent_increase = (df_temp.iloc[0] - df_temp.iloc[-1]) / df_temp.iloc[-1] 
    print country," = ", percent_increase

Austria  =  -0.524348810872
Belgium  =  -0.191808191808
Cyprus  =  -0.128022759602
Germany  =  -0.0978593272171
Estonia  =  0.836322869955
Spain  =  -0.148919753086
Finland  =  -0.376459143969
France  =  -0.450045829514
Greece (GR)  =  -0.621646623497
Ireland  =  -0.169992019154
Italy  =  -0.316441441441
Lithuania  =  0.149700598802
Luxembourg  =  0.138364779874
Latvia  =  1.77734877735
Malta  =  -0.0355113636364
Netherlands  =  -0.216397849462
Portugal  =  0.697777777778
Slovenia  =  0.6
Slovakia  =  0.935754189944
Euro area (changing composition)  =  -0.2625250501


### 2) Which EU Country has the largest revolving loan % decrease from 2000 – 2016?

In [8]:
### Largest Percent Decrease -> Greece
for country in df: #Looping through each country
    df_temp = df[country].dropna() #drop NaN values for that country
    percent_decrease = (df_temp.iloc[-1] - df_temp.iloc[0]) / df_temp.iloc[-1]
    print country," = ", percent_decrease

Austria  =  0.524348810872
Belgium  =  0.191808191808
Cyprus  =  0.128022759602
Germany  =  0.0978593272171
Estonia  =  -0.836322869955
Spain  =  0.148919753086
Finland  =  0.376459143969
France  =  0.450045829514
Greece (GR)  =  0.621646623497
Ireland  =  0.169992019154
Italy  =  0.316441441441
Lithuania  =  -0.149700598802
Luxembourg  =  -0.138364779874
Latvia  =  -1.77734877735
Malta  =  0.0355113636364
Netherlands  =  0.216397849462
Portugal  =  -0.697777777778
Slovenia  =  -0.6
Slovakia  =  -0.935754189944
Euro area (changing composition)  =  0.2625250501


### 3) Qualitatively describe in 5 sentences or less why these 2 countries may be seen these big changes during this period.

a) Latvia: After a deep recession that originated in the wake of the global financial crisis, Latvia’s GDP grew 4.7% in 2011-2013 on average, making Latvia one of the fastest growing economies in the EU. Economic growth was above 2% in 2014 – 2015 (2.4 % in 2014 and 2.3 % in the 1st half of 2015). Economic activity has been mainly driven by private 
consumption and exports with limited contribution from investment. Inflation has stabilized since 2009, which is  important given Latvia’s joining the euro in January 2014.(There is big increase in latvia's revolving loans from 2014). Inflation has remained low (0% in 2013; 0.6% in 2014 and 0.5% in August 2015) due to global commodity price 
developments that outweighed influence of growing income. Unemployment has decreased to 9.8% in the 2nd Q of 2015, which is higher than the 7.7% level in 2008.

b) Greece: Greece revolving loan % is gradually decreasing since 2000. Greece’s economy was never strong enough to share a currency with Germany’s. Reckless lending by French and German banks allowed the Greeks to finance widening budget and current account deficits for six years, but private capital flows dried up sharply after the 2008 crisis, 
forcing Greece to seek help from Eurozone governments and the International Monetary Fund in 2010. All that was 6 years ago, but Greece has not managed to turn the corner since then, when every other Eurozone country that took a bailout has. The primary reason for this is that Greece was the first country to ask for help, and the Eurozone was totally unprepared for it on all levels. The bailout was badly conceived, focusing too much on the budget balance and not enough on fixing Greece’s uniquely dysfunctional state apparatus. In a normal recession, government spending can offset the negative effects of private demand contracting, but in this case, the budgetary austerity drove Greece into a vicious spiral. The economy contracted by 25% between 2010 and 2014, fatally weakening Greece’s ability ever to repay its debts.